cpp/turbodbc_arrow/Test/tests/arrow_result_set_test.cpp

```diff
@@ -220,23 +231,18 @@ TEST_F(ArrowResultSetTest, MultiBatchConversionInteger)
 
 TEST_F(ArrowResultSetTest, ConversionCompressedInteger)
 {
-    arrow::Int64Builder builder;
-    arrow::Int16Builder compressed_builder;
-    for (int64_t i = 0; i < 1024; i++) {
-      ASSERT_OK(builder.Append(i));
-      ASSERT_OK(compressed_builder.Append(i));
-    }
     std::shared_ptr<arrow::Array> array;
-    ASSERT_OK(builder.Finish(&array));
-    std::shared_ptr<arrow::Array> compressed_array;
-    ASSERT_OK(compressed_builder.Finish(&compressed_array));
-
-    expected_arrays.push_back(compressed_array);
-    expected_fields.push_back(arrow::field("int_column", arrow::int16(), true));
+    make_int_range<arrow::Int64Type>(1024, &array);
```

Comment:
Why don't you just return a new shared pointer that contains an array?

Reply:
Because GTEST assertions only work in functions that return void.